### PR TITLE
Fix eslint issue on chrome devtools

### PR DESCRIPTION
**Pull request by matthieu-crouzet**
## Proposed change
Fix eslint issue on chrome devtools

### CODE REVIEW

File: apps/chrome-devtools/src/app-devtools/component-panel/component-panel-pres.template.html [hunk 1]
```diff
@@ -1,6 +1,10 @@
 @if (isComponentSelectionAvailable$ | async) {
   <div class="d-flex p-2 border-bottom align-items-center gap-2">
-    <i (click)="toggleInspector()" class="fa-external-link-square" [class.text-primary]="inspectorRunning"></i>
+    <i class="fa-external-link-square" [class.text-primary]="inspectorRunning"
+      (click)="toggleInspector()"
+      (keyup.enter)="toggleInspector()"
```
Comment (fpaul-1A) [inline apps/chrome-devtools/src/app-devtools/component-panel/component-panel-pres.template.html:5]: is it really needed ?

File: apps/chrome-devtools/src/extension/background.ts [hunk 2]
```diff
@@ -18,6 +32,7 @@ const WHITELISTED_HOSTS_KEY: typeof WhitelistedHostsKey = 'WHITELISTED_HOSTS';
  * @param message
  */
 const isOtterDebugMessage = (message: any): message is OtterMessage => {
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access -- message is of type `any`
   return message?.type === postMessageType;
```
Comment (fpaul-1A) [inline apps/chrome-devtools/src/extension/background.ts:36]: wouldn't this be cleaner ?
```suggestion
  return (message as OtterMessage)?.type === postMessageType;
```

File: apps/chrome-devtools/src/extension/background.ts [hunk 3]
```diff
@@ -72,33 +86,35 @@ const applyActivateState = async (appName: string, tabId: number) => {
     await chrome.scripting.executeScript({
       target: { tabId },
       args: [dataType, JSON.stringify(content), postMessageType],
-      // eslint-disable-next-line prefer-arrow/prefer-arrow-functions, @typescript-eslint/no-shadow
-      func: function (dataType, content, postMessageType) {
+      // eslint-disable-next-line prefer-arrow/prefer-arrow-functions -- expecting fat function to handle the context correctly
```
Comment (fpaul-1A) [inline apps/chrome-devtools/src/extension/background.ts:89]: unless I'm missing something, we are not actually using the context (`this`) in this function

File: eslint.shared.config.mjs [hunk 4]
```diff
@@ -123,7 +123,20 @@ export default [
...
+      '@stylistic/quote-props': 'off',
+      'jsdoc/require-param-description': 'off'
+    }
+  },
+  {
+    name: '@o3r/components',
+    files: ['**/*.component.ts'],
+    rules: {
+      // TODO re-enable this rule once it support arrow function in `computed` Angular signal computation
+      // No issue open on their side for the
```
Comment (fpaul-1A) [inline eslint.shared.config.mjs:135]:
```suggestion
      // No issue open on their side yet
```

File: apps/chrome-devtools/src/app-devtools/localization-panel/localization-panel-pres.template.html [hunk 5]
```diff
@@ -59,13 +59,15 @@ <h3 ngbAccordionHeader>
...
-                            activeStateName() && localizationActiveStateOverridesForCurrentLang()[localization.key] !== localizationLocalStateOverridesForCurrentLang()[localization.key]
-                              ? 'Reset value to the one from ' + activeStateName() + ': ' + (localizationActiveStateOverridesForCurrentLang()[localization.key] || localization.value)
-                              : 'Reset value to the default one from metadata: ' + localization.value
-                          "
-                          (click)="onLocalizationReset(localization)">
+                          [ngbTooltip]="text"
+                          [attr.aria-label]="text"
+                          tabindex="0"
+                          (click)="onLocalizationReset(localization)"
+                          (keyup.enter)="onLocalizationReset(localization)">
```
Comment (cpaulve-1A) [inline apps/chrome-devtools/src/app-devtools/localization-panel/localization-panel-pres.template.html:70]: same as mentioned by flo in https://github.com/AmadeusITGroup/otter/pull/2316/files#r1808617784

File: apps/chrome-devtools/src/app-devtools/theming-panel/theming-panel-pres.template.html [hunk 6]
```diff
@@ -28,7 +28,7 @@
           <h3 ngbAccordionHeader>
             <button ngbAccordionButton>
               @if (group.isPalette && group.defaultVariable) {
-                <div (click)="$event.stopPropagation();">
+                <div tabindex="-1" (click)="$event.stopPropagation();" (keyup.enter)="$event.stopPropagation();">
```
Comment (cpaulve-1A) [inline apps/chrome-devtools/src/app-devtools/theming-panel/theming-panel-pres.template.html:31]: What the $event.stopPropagation for here? The button is not a type=submit, is there really an event to stop propagating? (Actually why is the button on the whole html block and not around the label?)

File: apps/chrome-devtools/src/app-devtools/theming-panel/theming-panel-pres.template.html [hunk 7]
```diff
@@ -103,7 +105,9 @@ <h3 ngbAccordionHeader>
                                     ? 'Reset value to the one from ' + activeStateName() + ': ' + (themingActiveStateOverrides()[variable.name] || variable.defaultValue)
                                     : 'Reset value to the default one from metadata: ' + variable.defaultValue
                                 "
-                                (click)="onColorReset(variable)">
+                                tabindex="0"
+                                (click)="onColorReset(variable)"
+                                (keyup.enter)="onColorReset(variable)">
```
Comment (cpaulve-1A) [inline apps/chrome-devtools/src/app-devtools/theming-panel/theming-panel-pres.template.html:110]: Same remark as https://github.com/AmadeusITGroup/otter/pull/2316/files#r1808617784

File: apps/chrome-devtools/src/extension/background.ts (hunk 3 above)
Comment (cpaulve-1A) [reply apps/chrome-devtools/src/extension/background.ts:89]: Nor we use an arrow function?

File: apps/chrome-devtools/src/services/state.service.ts [hunk 8]
```diff
@@ -57,7 +77,7 @@ export class StateService {
     effect(async () => {
       const key = this.activeStateNameStorageKey();
       if (key) {
-        const name = (await chrome.storage.sync.get(key))[key] as string | undefined;
+        const { [key]: name } = (await chrome.storage.sync.get(key)) as Record<string, string | undefined>;
```
Comment (cpaulve-1A) [inline apps/chrome-devtools/src/services/state.service.ts:80]: Does it really work?
image

File: apps/chrome-devtools/src/app-devtools/theming-panel/theming-panel-pres.template.html (hunk 6 above)
Comment (matthieu-crouzet) [reply apps/chrome-devtools/src/app-devtools/theming-panel/theming-panel-pres.template.html:32]: I added a comment above to explain why we need to stop the propagation

File: apps/chrome-devtools/src/extension/wrap.ts [hunk 9]
```diff
@@ -13,33 +17,42 @@ window.postMessage({
...
 }
 
 /**
  * Listener for messages from the page
  * @param event
  */
-// eslint-disable-next-line prefer-arrow/prefer-arrow-functions
-(globalThis as any).localMessageListener = function (event: MessageEvent<any>) {
+
+// eslint-disable-next-line prefer-arrow/prefer-arrow-functions -- expecting fat function to handle the context correctly
```
Comment (fpaul-1A) [inline apps/chrome-devtools/src/extension/wrap.ts:34]: I don't see why we couldn't use arrow function here, we don't make use of the context

Comment (kpanot) [reply apps/chrome-devtools/src/extension/wrap.ts]: I think `window.addEventListener` enforce fat function

File: eslint.shared.config.mjs [hunk 10]
```diff
@@ -108,7 +108,7 @@ export default [
   },
   {
     name: '@o3r/eslint-config',
-    files: ['**/eslint*.config.mjs'],
+    files: ['**/*.mjs'],
```
[the remark this answers is not in the document]
Comment (matthieu-crouzet) [reply eslint.shared.config.mjs:111]: I forgot to change the name of the group
but with mjs you don't have access to `__dirname` and `__filename`, you have to code them 
at the beginning I did it only for eslint config but it's indeed linked to all mjs files

File: eslint.shared.config.mjs [hunk 11]
```diff
@@ -123,7 +123,20 @@ export default [
...
-      '@stylistic/quote-props': 'off'
+      '@stylistic/quote-props': 'off',
+      'jsdoc/require-param-description': 'off'
+    }
+  },
+  {
+    name: '@o3r/components',
+    files: ['**/*.component.ts'],
+    rules: {
+      // TODO re-enable this rule once it support arrow function in `computed` Angular signal computation
```
[the remark this answers is not in the document]
Comment (matthieu-crouzet) [reply eslint.shared.config.mjs:134]: yes I can move it there

File: apps/chrome-devtools/src/extension/wrap.ts [hunk 12]
```diff
@@ -13,33 +17,44 @@ window.postMessage({
...
-// eslint-disable-next-line prefer-arrow/prefer-arrow-functions
-(globalThis as any).localMessageListener = function (event: MessageEvent<any>) {
+function messageListener(event: MessageEvent<any>) {
   let message: any;
+  /* eslint-disable @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access -- message is of type `any` */
   try {
     message = typeof event.data === 'string' ? JSON.parse(event.data) : event.data;
   } catch (e) {
     // ignore invalid JSON
+    // eslint-disable-next-line no-console -- needed to warn the user
```
Comment (kpanot) [inline apps/chrome-devtools/src/extension/wrap.ts:40]: could be disabled at file level, this is specific to this file that is injected into a page and cannot have additional dependencies

File: apps/chrome-devtools/src/extension/background.ts [hunk 13]
```diff
@@ -114,12 +128,22 @@ const applyActivateState = async (appName: string, tabId: number) => {
 /** map of connection base on Tab ID */
 const connections = new Map<number, chrome.runtime.Port[]>();
 
+function postMessageToEveryone(content: string, type: typeof postMessageType) {
+  window.postMessage({
+    type: type,
+    to: 'app',
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment -- type of `content` is explicitly `any`
```
Comment (kpanot) [inline apps/chrome-devtools/src/extension/background.ts:135]: Maybe you should think about removing the `@typescript-eslint/no-unsafe-assignment` rule, regarding the number of exceptions in this single small package :S

**Review by fpaul-1A — APPROVED** [review]

**Review by kpanot — APPROVED** [review]

**Pull request merged**